Components/CANRx/CanRxTask.hpp

```diff
@@ -0,0 +1,37 @@
+/**
+ ******************************************************************************
+ * File Name          : CANTxTask.hpp
+ * Description        :
+ ******************************************************************************
+*/
+#include "Task.hpp"
+#include "main.h"
+#include "Queue.hpp"
+#include "Mutex.hpp"
+#include "CAN.h"
+
+enum CAN_RX_COMMANDS {
+    CAN_INTERRUPT_HAPPENED, //Task specific command queued on CAN_INT ISR
+};
+class CANRxTask : public Task
+{
+public:
+    static CANRxTask& Inst() {
+        static CANRxTask inst;
+        return inst;
+    }
+
+    void InitTask();
+    Queue* GetCAN_RX_QUEUE() const { return CAN_RX_QUEUE; }
+
+protected:
+    static void RunTask(void* pvParams) { CANRxTask::Inst().Run(pvParams); } // Static Task Interface, passes control to the instance Run();
+    void Run(void * pvParams); // Main run code
+    void HandleCommand(Command& cm);
+
+private:
+    CANRxTask();        // Private constructor
+    CANRxTask(const CANRxTask&);                        // Prevent copy-construction
+    CANRxTask& operator=(const CANRxTask&);            // Prevent assignment
+    Queue* CAN_RX_QUEUE;
```

Comment:
Is this queue actually getting allocated? Recommend using the qEvtQueue for the primary task queue (the queue the task blocks on), since all interfaces in Task.hpp point to that